presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestReorderJoins.java

```diff
@@ -0,0 +1,262 @@
+                .withStats(ImmutableMap.of(
+                        new PlanNodeId("valuesA"), PlanNodeStatsEstimate.builder()
+                                .setOutputRowCount(10000)
+                                .setSymbolStatistics(ImmutableMap.of(new Symbol("A1"), new SymbolStatsEstimate(0, 100, 0, 640000, 100)))
+                                .build(),
+                        new PlanNodeId("valuesB"), PlanNodeStatsEstimate.builder()
+                                .setOutputRowCount(10000)
+                                .setSymbolStatistics(ImmutableMap.of(new Symbol("B1"), new SymbolStatsEstimate(0, 100, 0, 640000, 100)))
+                                .build()))
+                .matches(join(
+                        INNER,
+                        ImmutableList.of(equiJoinClause("A1", "B1")),
+                        Optional.empty(),
+                        Optional.of(PARTITIONED),
+                        values(ImmutableMap.of("A1", 0)),
+                        values(ImmutableMap.of("B1", 0))
+                ));
+    }
+
+    @Test
+    public void testReplicatesWhenRequiredBySession()
+    {
+        tester.assertThat(new ReorderJoins(new CostComparator(1, 1, 1)))
+                .on(p ->
+                        p.join(
+                                INNER,
+                                p.values(new PlanNodeId("valuesA"), p.symbol("A1", BIGINT)),
+                                p.values(new PlanNodeId("valuesB"), p.symbol("B1", BIGINT)),
+                                ImmutableList.of(new JoinNode.EquiJoinClause(p.symbol("A1", BIGINT), p.symbol("B1", BIGINT))),
+                                ImmutableList.of(p.symbol("A1", BIGINT), p.symbol("B1", BIGINT)),
+                                Optional.empty()))
+                .setSystemProperty("join_distribution_type", "REPLICATED")
+                .withStats(ImmutableMap.of(
+                        new PlanNodeId("valuesA"), PlanNodeStatsEstimate.builder()
+                                .setOutputRowCount(10000)
+                                .setSymbolStatistics(ImmutableMap.of(new Symbol("A1"), new SymbolStatsEstimate(0, 100, 0, 640000, 100)))
+                                .build(),
+                        new PlanNodeId("valuesB"), PlanNodeStatsEstimate.builder()
+                                .setOutputRowCount(10000)
+                                .setSymbolStatistics(ImmutableMap.of(new Symbol("B1"), new SymbolStatsEstimate(0, 100, 0, 640000, 100)))
+                                .build()))
+                .matches(join(
+                        INNER,
+                        ImmutableList.of(equiJoinClause("A1", "B1")),
+                        Optional.empty(),
+                        Optional.of(REPLICATED),
+                        values(ImmutableMap.of("A1", 0)),
+                        values(ImmutableMap.of("B1", 0))
+                ));
+    }
+
+    @Test
+    public void testDoesNotFireWithNoStats()
```

Comment:
This test is failing because the cost calculator doesn't properly set unknown cost with no stats.  Values nodes always return 0 cost, but tablescans don't handle unknown stats correctly either.